pkg/ccl/streamingccl/streamclient/client_test.go

```diff
@@ -138,8 +144,48 @@ func TestGetFirstActiveClientEmpty(t *testing.T) {
 
 }
 
+func TestExternalConnectionClient(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+
+	srv, db, _ := serverutils.StartServer(t, base.TestServerArgs{
+		DefaultTestTenant: base.TestControlsTenantsExplicitly})
+	defer srv.Stopper().Stop(ctx)
+
+	sql := sqlutils.MakeSQLRunner(db)
+	pgURL, cleanupSinkCert := sqlutils.PGUrl(t, srv.AdvSQLAddr(), t.Name(), url.User(username.RootUser))
+	defer cleanupSinkCert()
+
+	externalConnection := "replication-source-addr"
+	sql.Exec(t, fmt.Sprintf(`CREATE EXTERNAL CONNECTION "%s" AS "%s"`,
+		externalConnection, pgURL.String()))
```

Comment:
Heh. I was going to suggest using placeholders here, but placeholders here are very annoying to get working, at least they seem to be in the sql shell.